### PR TITLE
Add a simple Makefile to check for inconsistencies

### PR DESCRIPTION
**Pull request by palmer-dabbelt**
This checks to ensure the bindings that are in Linux match those that
are in here.

### CODE REVIEW

**Comment by neuschaefer** [comment]
Oh, I'm already working on a more complicated solution: https://github.com/neuschaefer/riscv-device-tree-doc/blob/dev/scripts/check-against-linux.py

**Comment by palmer-dabbelt** [comment]
Works for me :)

**Comment by neuschaefer** [comment]
I'll make a pull req in a few minutes

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+/check
+*.swp
```
Comment (neuschaefer) [inline .gitignore:2]: This makes sense on its own

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[submodule "linux"]
+	path = linux
+	url = git://github.com/torvalds/linux.git
```
Comment (neuschaefer) [inline .gitmodules:3]: I'd rather prefer to have a submodule here, especially such a big one. For local checks I'll use pre-existing my local linux tree anyway.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+check: $(patsubst %,check/%.diff,$(BINDINGS))
+check: check/kernel_version
+
+check/kernel_version: .git/modules/linux/refs/heads/master
+	if [ $$(date -d "last week" +%s) -gt $$(git -C linux show --format=%at | head -n1) ]; then echo "Linux is too old"; exit 1; fi
+
+check/%.diff: bindings/% linux/Documentation/devicetree/bindings/%
+	diff -u $^
+
+check/%.diff: bindings/%
```
Comment (neuschaefer) [inline Makefile:12]: This does actually look quite nice and short… but I think a python script is more extensible in the long run

File: .gitignore (hunk 1 above)
Comment (neuschaefer) [reply .gitignore:2]: (the `*.swp`)

**Pull request closed without merging**